cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/datastore/Database200MigrationTest.java

```diff
@@ -0,0 +1,530 @@
+/**
+ * Copyright (C) 2016 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.sync.datastore;
+
+import static org.hamcrest.Matchers.is;
+
+import com.cloudant.mazha.Document;
+import com.cloudant.sync.datastore.migrations.MigrateDatabase100To200;
+import com.cloudant.sync.sqlite.Cursor;
+import com.cloudant.sync.sqlite.SQLDatabase;
+import com.cloudant.sync.sqlite.SQLDatabaseQueue;
+import com.cloudant.sync.sqlite.SQLQueueCallable;
+import com.cloudant.sync.util.DatabaseUtils;
+import com.cloudant.sync.util.TestUtils;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Created by rhys on 15/08/2016.
+ */
+public class Database200MigrationTest {
+
+    private Datastore ds;
```

Comment:
Why not just make this a DatastoreImpl and do the cast once in setUp, I think it will reduce the clutter.